appium/lib/screen-object.rb

```diff
@@ -66,84 +69,142 @@ def wait_step(timeout = 5, message = nil, &block)
+          when :down then  [end_x * 0.5, (end_y - (height * 0.2)), start_x * 0.5, (start_y + (height * 0.3)), duration]
+          when :left then  [end_x * 0.9,  end_y - (height / 2), start_x, end_y - (height / 2), duration]
+          when :right then [end_x * 0.1, end_y - (height / 2), end_x * 0.9, end_y - (height / 2), duration]
+          else
+            raise('Only up, down, left and right scrolling are supported')
+          end
+    gesture(loc)
+  end
 
-  def scroll_up_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
-        false
-      end
+  # Scroll Down in a direction until a string that matches is found,
+  #
+  # @param text      [String] The text you are looking for on the screen
+  def scroll_down_to_text(text)
+    scroll_text_to_view(text)
+  end
+
+  # Scroll Up in a direction until a string that matches is found,
+  #
+  # @param text      [String] The text you are looking for on the screen
+  def scroll_up_to_text(text)
+    scroll_text_to_view(text, :up)
+  end
+
+  # Scrolls in a direction until a string that matches is found,
+  #
+  # @param text      [String] The text you are looking for on the screen
+  # @param direction [symbol] The direction to search for an string
+  # @param timeout   [Integer] The amount of times in seconds we want to scroll to find the element
+  # @return          [Boolean]
+  def scroll_text_to_view(text, direction = :down, timeout = 40)
+    wait_until(timeout,'Unable to find element',&->{text_visible?(text, direction)})
+  end
+
+  # Scrolls in a direction if a text that matches is not found. return false,  otherwise return true
+  #
+  # @param text   [String] The text you are looking for on the screen
+  # @param direction [symbol] The direction to search for an string
+  # @return          [Boolean]
+  def text_visible?(text, direction)
+    if driver.find(text).displayed?
+      scroll(direction)
+      true
+    else
+      scroll(direction)
+      false
     end
+  rescue Selenium::WebDriver::Error::NoSuchElementError
+    scroll(direction)
+    false
   end
 
+  # Scrolls in a direction if an element that matches is not found. return false,  otherwise return true
+  #
+  # @param element   [Element] The text you are looking for on the screen
+  # @param direction [symbol] The direction to search for an string
+  # @return          [Boolean]
+  def element_visible?(element={}, direction = :down)
```

Comment:
Since, this one is specific to an element, keep this method in element.rb and remive it from here. You can call it like: module `ScreenObject::AppElements::Element.element_visible?`